ui/apps/platform/src/Containers/SystemConfig/Details/components/RedHatLayeredProductsCard.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import {
+    Button,
+    Card,
+    CardBody,
+    CardTitle,
+    CodeBlock,
+    Divider,
+    Modal,
+    Stack,
+    StackItem,
+    Text,
+} from '@patternfly/react-core';
+
+import useModal from 'hooks/useModal';
+import { PlatformComponentRule } from 'types/config.proto';
+
+export type RedHatLayeredProductsCardProps = {
+    rule: PlatformComponentRule | undefined;
+};
+
+function RedHatLayeredProductsCard({ rule }: RedHatLayeredProductsCardProps) {
+    const { isModalOpen, openModal, closeModal } = useModal();
```

Comment:
Tiny nit: I had a PR a while back to consolidate the `useModal()` and similar hooks into a `useToggle()`, but it must have been closed due to other priorities. Either way, what do you think about a plain old `useState<boolean>` here instead? It doesn't look like `useModal()` gives us any other functionality other than setting a boolean flag.